### ISSUE TITLE
Feature Request: Background positioning for Slides Widget

### ISSUE THREAD

**Issue by simonclay** [pojome/elementor#4263]
Often, on different slides you need to have a different background position for different images.

It would be very helpful to have a dropdown list like we have on 'Section' backgrounds, e.g.. Top Left, Top Center, Top Right etc.

Or perhaps the ability to give individual slides a class. Then we could target them with some custom CSS.

Thanks! 😁

**Comment by Turanligil** [comment]
yes this would be very nice... its frustrating to crop all the images to get the right image position in the slider....

**Comment by simonclay** [comment]
Absolutely @Turanligil, not something a client/editor is easily able to do.

**Comment by tomas-eklund** [comment]
Agreed. I came here today to request this feature. We need a way to adjust the background-position (X and Y separately) both by keyword values (center, top, bottom, left, right), by percentage and possibly unit-based values. Individually, per image, of course.

Of course, one can do this using Custom CSS, but this is not so user-friendly:

```
selector .slick-slide:nth-child(1) .slick-slide-bg {
    background-position-y: 5%;
}
```

While you're at it, being able to add a background shadow to the text in the slider makes it much more readable on top of images. So please add that as well.

```
selector {
    text-shadow: 1px 1px 4px rgba(0,0,0,1.0);
}
```

**Comment by jcmatoskx** [comment]
> selector .slick-slide:nth-child(1) .slick-slide-bg {
>     background-position-y: 5%;
> }

I tried this but unfortuantely it doesnt seem  to work :(  I tried a few different tweaks like adding !important, a dot to .selector, using margin-right instead of background position but nothing seemed to affect the background position

**Comment by tomas-eklund** [comment]
> The code doesnt seem to work for slides with more than 1 slide :(.

If you're not into coding custom CSS I understand that my simple example didn't help you much. You have to actually learn CSS to use Custom CSS effectively. It also helps if you are comfortable using the Developer Tools (typically accessed by pressing F12) of your favorite browser to experiment with and tweak the CSS before actually putting it into Elementor. That's why I'm asking for this feature to be implemented, so that my clients can do this themselves without knowing CSS.

It should be possible to position the background image using `background-position` even if the size property is set to `contain`. But most probably, you will want to use the `background-position-x` (horizontal positioning) property rather than `background-position-y` (vertical), at least on desktop. I don't think changing the `margin` will be of much use. Although, naturally, I don't know exactly what kind of effect you are trying to accomplish, so maybe `margin` is right for you.

If you have more than one image, and you need to change the background-position on all of them, you need to target each image individually by changing the number inside `nth-child(n)`:

```
selector .slick-slide:nth-child(1) .slick-slide-bg {
    /* First slide. Assuming this slide image has size: cover */
    background-position-y: top; /* makes sure you see the top of the image */
}

selector .slick-slide:nth-child(2) .slick-slide-bg {
    /* Second slide: Assuming this slide image has size: contain */
    background-position-x: right; /* pushes image all the way to the right */
}

```

Now, if you need to alter the background position for tablet and mobile, the CSS becomes more 
complicated. It can be done, but it's quite a lot of custom code and it's not fun to maintain.

> ... tried a few different tweaks like adding !important, a dot to .selector

You should not put a dot on the `selector` keyword. The keyword is just a placeholder for the actual element that Elementor will eventually render and will be dynamically replaced with a long string of CSS selectors (something like `.elementor-2 .elementor-element.elementor-element-6bf499f .elementor-repeater-item-40095a5`) once the element and it's corresponding CSS is being rendered.

**Comment by jcmatoskx** [comment]
Thank you very much for helping me out and furthering my CSS knowledge along with it. I really, truly appreciate the attention given! Much love!

**Comment by knireis** [comment]
I definitely need this feature!

**Comment by simonclay** [comment]
Thanks for great suggestions. I really feel that for ease of client use, I can't ask them to insert code. Really hoping that this feature will be incorporated into Elementor.

**Comment by VenkmanNL** [comment]
+1 for this feature!

**Comment by semmtex** [comment]
FYI: The newest update changed the CSS class to .swiper-slide-bg in case people are wondering why their CSS no longer works. :)

**Comment by tomas-eklund** [comment]
Not only did they change the class names, they also changed the HTML structure significantly. 

In a slider with two slides they now render six elements by duplicating slide 1 and 2 three times. I have no idea why. Pseudo HTML to illustrate:
```
<div class="swiper-wrapper">
  <div class="swiper-slide" data-swiper-slide-index="0"><!-- slide 1 --></div>
  <div class="swiper-slide" data-swiper-slide-index="1"><!-- slide 2 --></div>
  <div class="swiper-slide" data-swiper-slide-index="0"><!-- slide 1 --></div>
  <div class="swiper-slide" data-swiper-slide-index="1"><!-- slide 2 --></div>
  <div class="swiper-slide" data-swiper-slide-index="0"><!-- slide 1 --></div>
  <div class="swiper-slide" data-swiper-slide-index="1"><!-- slide 2 --></div>
</div>
```

So now it's impossible to target slide number 2 with just the selector `.swiper-slide:nth-child(2)` since `nth-child(4)` and `nth-child(6)` are also showing slide number 2. Instead you have to target the `data-swiper-slide-index` attribute and remember that indexing starts from 0 and not from 1, so slide number 1 has index 0, slide number 2 has index 1 etc.

This updated custom CSS code adjusts the vertical background position of the first slide:
```
selector .swiper-slide[data-swiper-slide-index="0"] .swiper-slide-bg {
    background-position-y: 5%;
}
```

This just illustrates the need for the original reature request: a way to adjust background position in the GUI for the various screen sizes. Maintaining custom CSS like this is a nightmare, expecially when the Elementor developers changes things around as much as they do.

**Comment by simonclay** [comment]
Thanks for the heads-up @tomas-eklund. That's quite alarming.

**Comment by ndstate** [comment]
Any update on this? This is basic functionality that is missing.

**Comment by wwdes** [comment]
Can this widget be considered responsive without this functionality? Image positioning (or css background-position) is important because it lets one decide what part of an image is focused when page translates to devices with smaller viewports. Elementor is awesome, but here and there are small things like this that need attention.

**Comment by jonburdon** [comment]
Huge thanks to Thomas above for the solution.

Elementor - Can se expect this feature to be added? It's pretty essential especially for mobile responsive settings. Images never look good on desktop AND mobile and this was quite a lot of code for a simple thing...

**Comment by MindwireAustralia** [comment]
+1 for this feature! Really need to be able to customise image position per slide and then alter for tablet and mobile as well.

**Comment by Michel9329** [comment]
I came across this topic because I'm having issues with the images in my slider with Elementor pro... 
I need to change the position of the background images, but I can't do it... I tried the code below, changing the number of the index but nothing happened...
Do you guys know any solutions?
I know CSS but I'm not able to change it, even if I always did it in other website (not with Elementor)
I'm desperate 😔 

> Not only did they change the class names, they also changed the HTML structure significantly.
> 
> In a slider with two slides they now render six elements by duplicating slide 1 and 2 three times. I have no idea why. Pseudo HTML to illustrate:
> 
> ```
> <div class="swiper-wrapper">
>   <div class="swiper-slide" data-swiper-slide-index="0"></div>
>   <div class="swiper-slide" data-swiper-slide-index="1"></div>
>   <div class="swiper-slide" data-swiper-slide-index="0"></div>
>   <div class="swiper-slide" data-swiper-slide-index="1"></div>
>   <div class="swiper-slide" data-swiper-slide-index="0"></div>
>   <div class="swiper-slide" data-swiper-slide-index="1"></div>
> </div>
> ```
> 
> So now it's impossible to target slide number 2 with just the selector `.swiper-slide:nth-child(2)` since `nth-child(4)` and `nth-child(6)` are also showing slide number 2. Instead you have to target the `data-swiper-slide-index` attribute and remember that indexing starts from 0 and not from 1, so slide number 1 has index 0, slide number 2 has index 1 etc.
> 
> This updated custom CSS code adjusts the vertical background position of the first slide:
> 
> ```
> selector .swiper-slide[data-swiper-slide-index="0"] .swiper-slide-bg {
>     background-position-y: 5%;
> }
> ```
> 
> This just illustrates the need for the original reature request: a way to adjust background position in the GUI for the various screen sizes. Maintaining custom CSS like this is a nightmare, expecially when the Elementor developers changes things around as much as they do.

**Comment by bouveng** [comment]
Another 3 year old issue. Please fix?

Br.

/J

**Comment by simonclay** [comment]
@tomas-eklund I can't get the latest css code to work at all now.

**Comment by tomas-eklund** [comment]
@simonclay If you have a real world example I could perhaps take a look at it, but I've tested on a fresh Elementor install (latest version) and the example custom CSS code still works:

Here the first CSS rule applies to the first slide (index=0) and the second rule applies to the second slide (index=1). If you have more slides, just create more rules and increase the index number on each rule.

```
selector .swiper-slide[data-swiper-slide-index="0"] .swiper-slide-bg {
    background-position: top left;
}

selector .swiper-slide[data-swiper-slide-index="1"] .swiper-slide-bg {
    background-position: 75% 25%;
}
```

**Comment by simonclay** [comment]
Thanks very much @tomas-eklund, I had to find another route to solve the issue, so don't have a link to send. But perhaps it was a theme or plugin conflict. I will test on a clean install when I can and report back.

Thanks again.

**Comment by yapofire** [comment]
+1 for this

So important to be able to set background position for mobile separately to desktop, Same for the styles options of the text too. Currently you cant have desktop center, mobile at top for example
